src/libs/kata-types/src/config/hypervisor/dragonball.rs

```diff
@@ -0,0 +1,185 @@
+// Copyright (c) 2019-2021 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::io::Result;
+use std::path::Path;
+use std::sync::Arc;
+use std::u32;
+
+use super::{default, register_hypervisor_plugin};
+use crate::config::default::MAX_DB_VCPUS;
+use crate::config::default::MIN_DB_MEMORY_SIZE;
+use crate::config::hypervisor::{
+    VIRTIO_BLK, VIRTIO_BLK_MMIO, VIRTIO_FS, VIRTIO_FS_INLINE, VIRTIO_PMEM,
+};
+use crate::config::{ConfigPlugin, TomlConfig};
+use crate::{eother, resolve_path, validate_path};
+
+/// Hypervisor name for qemu, used to index `TomlConfig::hypervisor`.
+pub const HYPERVISOR_NAME_DRAGONBALL: &str = "dragonball";
+
+/// Configuration information for dragonball.
+#[derive(Default, Debug)]
+pub struct DragonballConfig {}
+
+impl DragonballConfig {
+    /// Create a new instance of `DragonballConfig`.
+    pub fn new() -> Self {
+        DragonballConfig {}
+    }
+
+    /// Register the dragonball plugin.
+    pub fn register(self) {
+        let plugin = Arc::new(self);
+        register_hypervisor_plugin(HYPERVISOR_NAME_DRAGONBALL, plugin);
+    }
+}
+
+impl ConfigPlugin for DragonballConfig {
+    fn get_max_cpus(&self) -> u32 {
+        MAX_DB_VCPUS
+    }
+    fn get_min_memory(&self) -> u32 {
+        MIN_DB_MEMORY_SIZE
+    }
+    fn name(&self) -> &str {
+        HYPERVISOR_NAME_DRAGONBALL
+    }
+
+    /// Adjust the configuration information after loading from configuration file.
+    fn adjust_configuration(&self, conf: &mut TomlConfig) -> Result<()> {
```

Comment:
All the identifiers are very clearly specified but some of them are quite long. I'd consider shortening `configuration` to `config` as that's a very common abbreviation (you could even try `cfg` tbh).

Reply:
fixed